presto-orc/src/main/java/com/facebook/presto/orc/OrcWriterOptions.java

```diff
@@ -316,4 +151,184 @@ public String toString()
+            this.stripeMaxRowCount = stripeMaxRowCount;
+            this.rowGroupMaxRowCount = rowGroupMaxRowCount;
+            this.dictionaryMaxMemory = dictionaryMaxMemory;
+            this.maxStringStatisticsLimit = maxStringStatisticsLimit;
+            this.maxCompressionBufferSize = maxCompressionBufferSize;
+            this.compressionLevel = compressionLevel;
+            this.streamLayout = streamLayout;
+            this.integerDictionaryEncodingEnabled = integerDictionaryEncodingEnabled;
+        }
+
+        public Builder withStripeMinSize(DataSize stripeMinSize)
+        {
+            this.stripeMinSize = requireNonNull(stripeMinSize, "stripeMinSize is null");
+            return this;
+        }
+
+        public Builder withStripeMaxSize(DataSize stripeMaxSize)
+        {
+            this.stripeMaxSize = requireNonNull(stripeMaxSize, "stripeMaxSize is null");
+            return this;
+        }
+
+        public Builder withStripeMaxRowCount(int stripeMaxRowCount)
+        {
+            checkArgument(stripeMaxRowCount >= 1, "stripeMaxRowCount must be at least 1");
+            this.stripeMaxRowCount = stripeMaxRowCount;
+            return this;
+        }
+
+        public Builder withRowGroupMaxRowCount(int rowGroupMaxRowCount)
+        {
+            checkArgument(rowGroupMaxRowCount >= 1, "rowGroupMaxRowCount must be at least 1");
+            this.rowGroupMaxRowCount = rowGroupMaxRowCount;
+            return this;
+        }
+
+        public Builder withDictionaryMaxMemory(DataSize dictionaryMaxMemory)
+        {
+            this.dictionaryMaxMemory = requireNonNull(dictionaryMaxMemory, "dictionaryMaxMemory is null");
+            return this;
+        }
+
+        public Builder withMaxStringStatisticsLimit(DataSize maxStringStatisticsLimit)
+        {
+            this.maxStringStatisticsLimit = requireNonNull(maxStringStatisticsLimit, "maxStringStatisticsLimit is null");
+            return this;
+        }
+
+        public Builder withMaxCompressionBufferSize(DataSize maxCompressionBufferSize)
+        {
+            this.maxCompressionBufferSize = requireNonNull(maxCompressionBufferSize, "maxCompressionBufferSize is null");
+            return this;
+        }
+
+        public Builder withCompressionLevel(OptionalInt compressionLevel)
+        {
+            this.compressionLevel = requireNonNull(compressionLevel, "compressionLevel is null");
+            return this;
+        }
+
+        public Builder withStreamLayout(StreamLayout streamLayout)
+        {
+            this.streamLayout = requireNonNull(streamLayout, "streamLayout is null");
+            return this;
+        }
+
+        public Builder withIntegerDictionaryEncodingEnabled(boolean integerDictionaryEncodingEnabled)
+        {
+            this.integerDictionaryEncodingEnabled = integerDictionaryEncodingEnabled;
+            return this;
+        }
+
+        public DataSize getStripeMinSize()
```

Comment:
It's a little unusually to have access methods in a builder? Why are they needed?

Reply:
getter methods are removed.